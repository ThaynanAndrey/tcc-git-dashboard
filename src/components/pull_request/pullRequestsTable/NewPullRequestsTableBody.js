import React from 'react';
import Tooltip from "react-simple-tooltip";

/**
 * Stateless component with the function of presenting the body of
 * a table with Pull Requests not yet added to the project.
 * 
 * @author Thaynan Nunes
 */
const NewPullRequestsTableBody = ({ pullRequests, addPullRequest }) => {
    const pullRequestElements = pullRequests.map((pullRequest, indice) => (
        <tr key={indice}>
            <td>{ pullRequest.nome }</td>
            <td>{ pullRequest.dataCriacao }</td>
            <td>{ pullRequest.status }</td>
            <td>{ pullRequest.responsavel.nome }</td>
            <td>{ pullRequest.repositorio.nome }</td>
            <td>
                <Tooltip content="Adicionar PR" placement="left" radius={10} style={{whiteSpace: "nowrap"}}>
                    <i className="material-icons left green-text" style={{cursor: "pointer"}}
                        onClick={() => addPullRequest(pullRequest)}>
                        add_box
                    </i>
                </Tooltip>
            </td>
        </tr>
    ));

    return (
        <tbody>
            {pullRequestElements}
        </tbody>
    )
}

export default NewPullRequestsTableBody;