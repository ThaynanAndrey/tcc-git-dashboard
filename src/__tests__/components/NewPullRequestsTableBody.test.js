import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import NewPullRequestsTableBody from '../../components/pull_request/pullRequestsTable/NewPullRequestsTableBody';

describe('Test NewPullRequestsTableBody', () => {
  it('renders correctly a NewPullRequestsTableBody\'s component with correct props', () => {
    const pullRequests = [ { name: "Tests", dataCriacao: "10/10/1010", status: "open",
        responsavel: { nome: "Ze" }, repositorio: { nome: "Repo" } } ];
    const addPullRequest = jest.fn();

    const tree = mount(
      <NewPullRequestsTableBody
        pullRequests={pullRequests}
        addPullRequest={addPullRequest}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();

    expect(tree.prop('pullRequests')).toEqual(pullRequests);
    expect(tree.prop('addPullRequest')).toEqual(addPullRequest);
  });

  it('renders correctly a NewPullRequestsTableBody\'s component with correct classes', () => {
    const pullRequests = [ { name: "Tests", dataCriacao: "10/10/1010", status: "open",
        responsavel: { nome: "Ze" }, repositorio: { nome: "Repo" } } ];
    const addPullRequest = jest.fn();

    const tree = mount(
      <NewPullRequestsTableBody
        pullRequests={pullRequests}
        addPullRequest={addPullRequest}
      />
    );

    expect(tree.find('tbody').exists()).toBeTruthy();
    expect(tree.find('tr').exists()).toBeTruthy();
  });
})