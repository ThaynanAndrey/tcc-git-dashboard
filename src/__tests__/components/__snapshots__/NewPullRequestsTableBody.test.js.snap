// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NewPullRequestsTableBody renders correctly a NewPullRequestsTableBody's component with correct props 1`] = `
<NewPullRequestsTableBody
  addPullRequest={[MockFunction]}
  pullRequests={
    Array [
      Object {
        "dataCriacao": "10/10/1010",
        "name": "Tests",
        "repositorio": Object {
          "nome": "Repo",
        },
        "responsavel": Object {
          "nome": "Ze",
        },
        "status": "open",
      },
    ]
  }
>
  <tbody>
    <tr
      key="0"
    >
      <td />
      <td>
        10/10/1010
      </td>
      <td>
        open
      </td>
      <td>
        Ze
      </td>
      <td>
        Repo
      </td>
      <td>
        <Tooltip.Wrapper
          arrow={8}
          background="#000"
          border="#000"
          color="#fff"
          content="Adicionar PR"
          fadeDuration={0}
          fadeEasing="linear"
          fixed={false}
          fontFamily="inherit"
          fontSize="inherit"
          offset={0}
          padding={16}
          placement="left"
          radius={10}
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }
          zIndex={1}
        >
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "1oin9j2",
                "next": undefined,
                "styles": "
          
  position: relative;
  display: inline-block;
;
          
        ",
              }
            }
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "whiteSpace": "nowrap",
              }
            }
          >
            <div
              className="css-1oin9j2"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "whiteSpace": "nowrap",
                }
              }
            >
              <i
                className="material-icons left green-text"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                add_box
              </i>
              <Tooltip
                fadeDuration={0}
                fadeEasing="linear"
                offset={8}
                open={false}
                placement="left"
                zIndex={1}
              />
            </div>
          </EmotionCssPropInternal>
        </Tooltip.Wrapper>
      </td>
    </tr>
  </tbody>
</NewPullRequestsTableBody>
`;
